### ISSUE TITLE
VPAT Reader: focus

### ISSUE THREAD

**Issue by abaevbog** [zotero/zotero#4222]
- [x] vpat 61: Magnifying class of the "Search annotations" input should not be focusable
    - It could be focused with voiceover on arrow left click from the input.
 - [x] vpat 62: Clearing sidebar "Search annotation" resets focus
    - The review indicates "When this search is cleared with the button labeled 'x,' focus is moved to the beginning of the PDF viewer. This occurs with or without a screen reader active." Does not happen consistently for me but is sometimes reproducible on mac
 - [X] vpat 63: Sidebar unable to move focus forward if no results for search. Fixed?
 - [x] vpat 66 and 68: focus does not enter the tag selector popup for annotations
     - Once the tags popup is opened, arrows or tab keypresses will keep navigating the sidebar instead of the tags selector
 - [x] vpat 67: even if focus is places inside of the tag selector popup, Escape does not close it
     - Escape from tag selector popup should close it
  - [x] Something I noticed myself: tabbing through an annotation will move focus from the context menu button to the comment to the tags button skipping the actual text of the annotation unless it has been focused by selecting "Edit Annotation Text" in the context menu. It seems a bit inconsistent, perhaps annotation text should have a tab stop as well? Or maybe we announce the text of the annotation when the annotation box is focused? Otherwise, the only way to have annotation text announced via screen reader seems to be via "Edit Annotation Text" which is not very intuitive and has to be discovered.

**Comment by mrtcode** [comment]
> Something I noticed myself: tabbing through an annotation will move focus from the context menu button to the comment to the tags button skipping the actual text of the annotation unless it has been focused by selecting "Edit Annotation Text" in the context menu. It seems a bit inconsistent, perhaps annotation text should have a tab stop as well? Or maybe we announce the text of the annotation when the annotation box is focused? Otherwise, the only way to have annotation text announced via screen reader seems to be via "Edit Annotation Text" which is not very intuitive and has to be discovered.

Yes, I guess we can add a tab stop.

And the other issues seem to be fixed already, right?

**Comment by abaevbog** [comment]
> Yes, I guess we can add a tab stop.

Cool - I added a small [tweak](https://github.com/zotero/reader/pull/127/commits/27304beda799ae8b2c4e1f4099291a0e8db07115) to my accessibility PR to set `tabindex` and `data-tabstop` somewhat independently of `readonly` state. That way, we can have a focusable but still readonly annotation text. That is to preserve existing behavior of making annotation text both focusable AND editable on double-click.

> And the other issues seem to be fixed already, right?

The tag selector issues (vpat 66-68) should be all addressed in #4231 and the rest of the issues here and in #4220 should be covered by the reader PR.

Some parts of #4223 will also be addressed by the reader PR. But a couple of issues there (related to annotations and search results displayed in the actual scrollable reader area) seemed a bit trickier so I thought I should ask before trying to tackle those. I'll write up a separate comment there with more details for those issues
